IPython/lib/display.py

```diff
@@ -255,8 +255,12 @@ class YouTubeVideo(IFrame):
     """
 
     def __init__(self, id, width=400, height=300, **kwargs):
+        self.id=id
         src = "https://www.youtube.com/embed/{0}".format(id)
         super(YouTubeVideo, self).__init__(src, width, height, **kwargs)
+    def _repr_jpeg_(self):
```

Comment:
Also if you are doing modification can you pute a blank line in between the 2 functions ?